_targets.R

```diff
@@ -78,17 +77,17 @@ p2_targets_list <- list(
     process_data(site_data)
   ),
   tar_target(
-    site_data_annotated,
-    annotate_data(site_data_clean, site_filename = site_info_csv)
+    site_data_annotated_csv,
+    annotate_data(site_data_clean, site_info = site_info, out_filename = '2_process/out/site_data_annotated.csv'),
+    format = "file"
   )
 )
 
 p3_targets_list <- list(
   tar_target(
-    figure_1_png,
-    plot_nwis_timeseries(fileout = "3_visualize/out/figure_1.png", site_data_annotated,
-                         width = p_width, height = p_height, units = p_units),
-    format = "file"
+    figure_1,
+    plot_nwis_timeseries(site_data_annotated_csv = site_data_annotated_csv,
+                         width = p_width, height = p_height, units = p_units)
```

Comment:
Figures aren't as useful in pipelines when they aren't saved as files IMO. To see it you would need to run `tar_load(figure_1)` and then print `figure_1` in your console, vs just going to the file and looking. 

That said, I could see a use case where you might use a ggplot object with a basic plotting call as a target and then build plots that diverge from that initial shared ggplot() call. In that case. I would save those resulting figures as files not as objects.

For example, 
```r
tar_target(initial_ggplot, ggplot(data = mtcars, aes(x = cyl, y = mpg))),
tar_target(plot_by_disp_png, {
  gg_disp <- initial_ggplot + geom_point(aes(color = disp))
  ggsave("plot_by_disp.png", gg_disp)
}),
tar_target(plot_by_wt_png, {
  gg_wt <- initial_ggplot + geom_point(aes(color = wt))
  ggsave("plot_by_wt.png", gg_wt)
})
```

Reply:
Approved this because it works. Just noting that plots _**usually**_ are file targets, except for some edge cases.

Reply:
Yes, I agree it makes more sense as a file! Your example use case answers my question about the ggsave arguments. That's what I was imagining doing with the resulting object. Thanks!